### DISCUSSION TITLE
Allow filtering releases by package name (improve monorepo support)

### DISCUSSION THREAD

**Discussion by rossng** [community/community#5579] [Actions]
# The problem

In the JavaScript/TypeScript world, it's increasingly common for projects to have a single 'monorepo' from which multiple packages can be published independently. The version number may or may not be synchronised between packages.

I often need to review the release notes for the last few versions of a single package. For example, perhaps something has broken and I need to figure out when it might have happened. Or I am doing package upgrades and I want to scan through the changelog to identify any changes that might cause problems.

Unfortunately this is quite difficult with GitHub's Releases list today. Releases for all the different packages (often dozens!) are muxed together into a single list, which makes it really difficult to quickly retrieve the information I need.

# Try-at-home exercise

Looking at the [list of releases](https://github.com/atlassian/changesets/releases) for the Atlassian changesets project, find what changes were made in the last three releases of the `@changesets/cli` package.

# Some ideas

- Parse tag names that fit certain common formats. For example, `@<scope>/<package-name>@<semver>`
- Make it possible to filter the releases list by the parsed package name

# Related suggestions

https://github.com/github/feedback/discussions/5286

**Comment by ethomson** [comment]
This is good feedback - tagging @MylesBorins (who's been looking after releases _and_ has an interest in monorepos) so that he can capture this feedback.

**Reply by medmunds** [reply]
Here's another example: https://github.com/lit/lit/releases.

Even allowing search filters on a tag prefix or glob or exact substring would solve a lot of the problem: `tag:lit@*` or `tag:"lit@"` or similar.

**Comment by pavinduLakshan** [comment]
Is there any update on this? 👀

**Comment by f15u** [comment]
+1.

Watching releases for a repository that releases multiple packages (eg https://github.com/Effect-TS/effect/releases) it's impossible to find a **package** and see releases for it. 

A simple filter on the name lie `package-name@*` would be a massive improvement.

**Comment by nebuk89** [comment]
Hmmm good feedback and mono repo support is still something we need to improve. I will add this to our backlog/check it's there :)

**Comment by aqeelat** [comment]
This is badly needed. Another example is https://github.com/clerk/javascript/releases

**Comment by detj** [comment]
Core maintainer of [measure.sh](https://github.com/measure-sh/measure/releases). We release mobile SDKs for iOS/Android/KMP/React Native. A monorepo setup really helps us to keep everything together & in-sync. Though, when it comes to releases, finding/browsing the correct version becomes a discovery issue for our users.

Moreover, certain package registries, like the [Swift Package Index](https://swiftpackageindex.com/measure-sh/measure) only considers SemVer like versions, often picking the incorrect tag/release version.

If GitHub Releases supports something like [Release Channels](https://github.com/orgs/community/discussions/86745) natively - then other systems may also integrate with it better.

For now, we are contemplating on having different repos to get around this, but would really like to avoid doing that.
